src/include/hip/rpp_hip_common.hpp

```diff
@@ -72,4 +114,441 @@ inline RppStatus generate_gaussian_kernel_gpu(Rpp32f stdDev, Rpp32f* kernel, Rpp
+{
+    *((d_float8 *)(&dstPtr[dstIdx])) = *dst_f8;
+}
+
+// I8 loads and stores without layout toggle (8 I8 pixels)
+
+__device__ __forceinline__ void rpp_hip_load8_and_unpack_to_float8(signed char *srcPtr, uint srcIdx, d_float8 *src_f8)
+{
+    uint2 src = *((uint2 *)(&srcPtr[srcIdx]));
+    src_f8->x = rpp_hip_unpack(src.x);
+    src_f8->y = rpp_hip_unpack(src.y);
+}
+
+__device__ __forceinline__ void rpp_hip_pack_float8_and_store8(signed char *dstPtr, uint dstIdx, d_float8 *dst_f8)
+{
+    uint2 dst;
+    dst.x = rpp_hip_pack(dst_f8->x);
+    dst.y = rpp_hip_pack(dst_f8->y);
+    *((uint2 *)(&dstPtr[dstIdx])) = dst;
+}
+
+// F16 loads and stores without layout toggle (8 F16 pixels)
+
+__device__ __forceinline__ void rpp_hip_load8_and_unpack_to_float8(half *srcPtr, uint srcIdx, d_float8 *src_f8)
+{
+    d_half8 src_h8;
+    src_h8 = *((d_half8 *)(&srcPtr[srcIdx]));
+
+    *(float2 *)&(src_f8->x) = __half22float2(src_h8.x.x);
+    *((float2 *)&(src_f8->x) + 1) = __half22float2(src_h8.x.y);
+    *(float2 *)&(src_f8->y) = __half22float2(src_h8.y.x);
+    *((float2 *)&(src_f8->y) + 1) = __half22float2(src_h8.y.y);
+}
+
+__device__ __forceinline__ void rpp_hip_pack_float8_and_store8(half *dstPtr, uint dstIdx, d_float8 *dst_f8)
+{
+    d_half8 dst_h8;
+
+    dst_h8.x.x = __float22half2_rn(*(float2 *)&(dst_f8->x));
+    dst_h8.x.y = __float22half2_rn(*((float2 *)&(dst_f8->x) + 1));
+    dst_h8.y.x = __float22half2_rn(*(float2 *)&(dst_f8->y));
+    dst_h8.y.y = __float22half2_rn(*((float2 *)&(dst_f8->y) + 1));
+
+    *((d_half8 *)(&dstPtr[dstIdx])) = dst_h8;
+}
+
+// U8 loads with layout toggle PKD3 to PLN3 (24 U8 pixels)
+
+__device__ __forceinline__ void rpp_hip_load24_pkd3_and_unpack_to_float24_pln3(uchar *srcPtr, uint srcIdx, d_float24 *src_f24)
+{
+    d_uint6 src = *((d_uint6 *)(&srcPtr[srcIdx]));
+
+    src_f24->x.x = make_float4(rpp_hip_unpack0(src.x.x), rpp_hip_unpack3(src.x.x), rpp_hip_unpack2(src.x.y), rpp_hip_unpack1(src.y.x));
+    src_f24->x.y = make_float4(rpp_hip_unpack0(src.y.y), rpp_hip_unpack3(src.y.y), rpp_hip_unpack2(src.z.x), rpp_hip_unpack1(src.z.y));
+
+    src_f24->y.x = make_float4(rpp_hip_unpack1(src.x.x), rpp_hip_unpack0(src.x.y), rpp_hip_unpack3(src.x.y), rpp_hip_unpack2(src.y.x));
+    src_f24->y.y = make_float4(rpp_hip_unpack1(src.y.y), rpp_hip_unpack0(src.z.x), rpp_hip_unpack3(src.z.x), rpp_hip_unpack2(src.z.y));
+
+    src_f24->z.x = make_float4(rpp_hip_unpack2(src.x.x), rpp_hip_unpack1(src.x.y), rpp_hip_unpack0(src.y.x), rpp_hip_unpack3(src.y.x));
+    src_f24->z.y = make_float4(rpp_hip_unpack2(src.y.y), rpp_hip_unpack1(src.z.x), rpp_hip_unpack0(src.z.y), rpp_hip_unpack3(src.z.y));
+}
+
+// F32 loads with layout toggle PKD3 to PLN3 (24 F32 pixels)
+
+__device__ __forceinline__ void rpp_hip_load24_pkd3_and_unpack_to_float24_pln3(float *srcPtr, uint srcIdx, d_float24 *src_f24)
+{
+    src_f24->x.x.x = srcPtr[srcIdx];
```

Comment:
this function needs to be rewritten. please don't access the srcPtr memory like the way you've written here by adding some offset. read a float4 each time and then use the x, y, z, w components to fill the src_f24.

Reply:
Done